File: nodejsUdp/entity.js
var moment = require('moment');

var Map = function() {
  this.list = new Object();
};
Map.prototype = {
  put : function(key, value) {
    this.list[key] = value;
  },  
  get : function(key) {
    return this.list[key];
  },  
  remove : function(key) {
    delete this.list[key];
  },  
  exists : function(key) {
    return key in this.list;
  },  
  count : function() {
    return Object.keys(this.list).length;
  }
};

var User = function() {
  this.uid = "";
  this.group = "";
  this.public_ip = "";
  this.public_port = 0;
  this.heartbeat_time = moment();
};


var P2PGroup = function() {
  this.name = "";
  this.user_list = new Map();
};

exports.UserList = Map;
exports.User = User;
exports.P2PGroupList = Map;
exports.P2PGroup = P2PGroup;

File: csUdp/Chat.Common/Entity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Chat.Common
{
    public class User
    {
        public string uid;
        public string group;
        public string publicIp;
        public ushort publicPort;
    }

    public class Group
    {
        public string name;
        public Dictionary<string, User> userList = new Dictionary<string, User>();
    }

    public class Session
    {
        public User user = new User();
        public Group group = new Group();
        public bool isLogin = false;

        public void Clear()
        {
            user = new User();
            group = new Group();
            isLogin = false;
        }
    }
}
